### PR TITLE
Added authorisation capabilities

### PR DESCRIPTION
**Pull request by nmarquesantos**
In ConnectionImpl, for each handle method (handleSubscribe, handlePublisher, etc), I've put the core code inside a Consumer. It was just a matter of preference instead of using private functions, happy to change if we feel it is more structured and readable.

I only have doubts regarding the handling of unauthorised access. Returning an exception and error code is fine. However, I've noticed that the RequestID is required in most responses inside the ClientImpl class. Given that, I assumed it is correct to return that RequestID even though the user is unauthorised (or an exception due to authorisation is thrown)

### CODE REVIEW

File: src/main/java/com/tesco/mewbase/auth/Operation.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.tesco.mewbase.auth;
+
+public enum Operation {
+    SUBSCRIBE,
+    UNSUBSCRIBE,
+    PUBLISH,
+    UNPUBLISHgit
```
Comment (purplefox) [inline src/main/java/com/tesco/mewbase/auth/Operation.java]: typo?

Comment (nmarquesantos) [inline src/main/java/com/tesco/mewbase/auth/Operation.java]: yes ;) probably one of the moments ALT + Tab didn't work properly.

I'll amend it.

All ok with the rest?

File: src/main/java/io/mewbase/server/impl/auth/UnauthorizedUser.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public class UnauthorizedUser implements MewbaseUser {
+
+    @Override
+    public CompletableFuture<Boolean> isAuthorised(String protocolFrame) {
+        CompletableFuture<Boolean> unauthorisedCF = new CompletableFuture<>();
+
+        unauthorisedCF.complete(false);
+
+        return unauthorisedCF;
```
Comment (purplefox) [inline src/main/java/io/mewbase/server/impl/auth/UnauthorizedUser.java]: Can replace all of this with:

return CompletableFuture.completedFuture(false);

**Comment by purplefox** [comment]
Thanks! I did a bit of refactoring and merged this manually.

**Comment by nmarquesantos** [comment]
cheers @purplefox 👍

**Pull request merged**